Tiny contained change to oe_ memory functions

SeanTAllen opened the pull request.

letmaik commented:
@SeanTAllen This looks like it's not ready for review yet. Could you make it a draft PR?

SeanTAllen commented:
@davidchisnall can you have a look at this set of changes for #151? There are additional malloc/calloc/strdup/free calls to replace however, doing so causes some of the test suite to segfault. 

I spent some time looking at it and haven't found any obvious cause yet.

SeanTAllen committed:
Remove some libc usage in src/enclave

This commit removes some of the malloc/calloc/free/strdup libc usage
in src/enclave. Not all usage as been removed yet. There are other
cases that I have left because when changed, the test suite segfaults
for non-obvious reasons that need to be looked into.

This commit addresses a small part of #151.
